src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -1477,8 +1484,17 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
          */
         static constexpr uint32_t maxBuses = 128;
         std::array<float *, maxBuses> busses{};
+        std::array<double *, maxBuses> doubleBusses{};
         busses.fill(nullptr);
 
+        bool supportsDouble = processor->supportsDoublePrecisionProcessing();
+        bool hostCalledWithDouble = false;
+
+        if (supportsDouble)
+        {
+            doubleBusses.fill(nullptr);
```

Comment:
Neither this `fill` nor `busses.fill(nullptr)` should be necessary since both `busses` and `doubleBusses` are already zero-initialized with `{}`.

Reply:
So you are absolutely correct we init `busses` with `{}`

I thought we had not used the `{}` form, hence the fill, so conditionalized the double busses fill so it is unused.

I think our best bet is to remove the `{}` on double busses and keep the fill but it's not that much assembly difference. Basically I wanted case three here: https://godbolt.org/z/ev7TzrKGj but I accidentally wrote case 4!

Reply:
Sounds like a good idea.

For `busses`, zero-init with `{}` and no fill, and for `doubleBusses`, don't use zero-init and do a conditional fill. That would probably generate the best assembly.

While it isn't a huge assembly difference, it's inside the `process` method and calling `fill` is O(N), so it would be good to fix this.

Reply:
Yup that’s why I put the fill in an if. Just I didn’t see the {} above!